### PR TITLE
[docs] Marking internal comment in proofs docs

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

File: src/proof/proof_rule.h [hunk 1]
```diff
@@ -23,11 +23,14 @@
 namespace cvc5::internal {
 
 /**
+ * \internal
+ * This documentation is target for the online documentation that can
+ * be found at https://cvc5.github.io/docs/cvc5-1.0.0/proofs/proof_rules.html.
```
Comment (HanielB) [inline src/proof/proof_rule.h:28]: Actually, thinking about this, do we need to link with the version always?

Comment (nafur) [reply src/proof/proof_rule.h:28]: I'd suggest using https://cvc5.github.io/docs/latest/proofs/proof_rules.html

Comment (HanielB) [reply src/proof/proof_rule.h]: OK, done

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**